### ISSUE TITLE
Supporting per-file-ignores with indent + inline comments

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#776]
In GitLab by @jamesbraza on Oct 29, 2020, 11:57

*Please provide the exact, unmodified output of `flake8 --bug-report`*

<p>
<details>
<summary>`flake8 --bug-report`</summary>

```json
{
  "dependencies": [],
  "platform": {
    "python_implementation": "CPython",
    "python_version": "3.7.9",
    "system": "Darwin"
  },
  "plugins": [
    {
      "is_local": false,
      "plugin": "black",
      "version": "0.2.1"
    },
    {
      "is_local": false,
      "plugin": "flake8-docstrings",
      "version": "1.5.0, pydocstyle: 5.1.1"
    },
    {
      "is_local": false,
      "plugin": "flake8_isort",
      "version": "4.0.0"
    },
    {
      "is_local": false,
      "plugin": "mccabe",
      "version": "0.6.1"
    },
    {
      "is_local": false,
      "plugin": "pycodestyle",
      "version": "2.6.0"
    },
    {
      "is_local": false,
      "plugin": "pyflakes",
      "version": "2.2.0"
    }
  ],
  "version": "3.8.4"
}
```

</details>
</p>

*Please describe the problem or feature*

Inside of a `setup.cfg`, I like to use inline comments to explain certain parts.  For example, the [`ignore`](https://flake8.pycqa.org/en/latest/user/options.html#cmdoption-flake8-ignore) option:

```
ignore =
    D406,  # Google style requires ":" at end.
    D407,  # We aren't using numpy style
    D413,  # Blank line after last section. -> No blank line.
```

I also use the [`per-file-ignores`](https://flake8.pycqa.org/en/latest/user/options.html#cmdoption-flake8-per-file-ignores) option.  Currently, it looks like this, and works fine:

```
# D100, D103: testing doesn't always need docstrings
# F841: Having unassigned variables can be useful in test debugging
per-file-ignores =
    path/to/tests/*: D100, D103, F841,
```

I would like it to look like this:

- Indent for each code below `path/to/tests/*:`
- Comment explaining the reasoning for each code

```
per-file-ignores =
    path/to/tests/*:
        D100,  # Testing doesn't always need docstrings
        D103,  # Testing doesn't always need docstrings
        F841,  # Having unassigned variables can be useful in test debugging
```

However, when I run `flake8` with this configuration, I get this error message:

```
There was a critical error during execution of Flake8:
Expected `per-file-ignores` to be a mapping from file exclude patterns to ignore codes.

Configured `per-file-ignores` setting:

    path/to/tests/*:
    D100,  # Testing doesn't always need docstrings
    D103,  # Testing doesn't always need docstrings
    F841,  # Having unassigned variables can be useful in test debugging
```

Is it possible to add support for this method of specifying the `per-file-ignores` with indents + inline comments?

**Comment by asottile** [comment]
In GitLab by @jamesbraza on Oct 29, 2020, 12:01

changed the description

**Comment by asottile** [comment]
In GitLab by @jamesbraza on Oct 29, 2020, 12:03

changed the description

**Comment by asottile** [comment]
In GitLab by @jamesbraza on Oct 29, 2020, 12:04

changed title from **{-per-file-ignores with-} inline comments** to **{+Supporting per-file-ignores with indent ++} inline comments**

**Comment by asottile** [comment]
In GitLab by @asottile on Oct 29, 2020, 12:35

dupe of https://gitlab.com/pycqa/flake8/-/issues/682

see docs: https://flake8.pycqa.org/en/latest/user/configuration.html#project-configuration (the note at the bottom)

tl;dr:
- the recommend setting of `configparser` is what we're already using
- ini itself does not have inline comments
- enabling the setting in configparser would be a breaking change for flake8
- adding special casing for per-file-ignores would make it inconsistent with the rest of the configuration parsing
